src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -58,7 +58,12 @@ public override SemanticTokensOrSemanticTokensEdits GetSemanticTokensEdits(Razor
             }
             var newTokens = ConvertSemanticRangesToSemanticTokens(semanticRanges, codeDocument);
 
-            var semanticEdits = SyntaxTokenToSemanticTokensEditHelper.ConvertSyntaxTokensToSemanticEdits(newTokens, previousResults);
+            if (previousResults is null)
```

Comment:
What if I convert it to a conditional expression instead? `previousResultId == null` isn't the only way to get `previousResults == null`, it also happens if the `previousResult` was evicted from the cache (most likely scenario for that is if someone was working on a different document for a while).

Reply:
Ah. Didn't realize `_semanticTokensCache.Get(previousResultId);` could return null. In case you touch it in the future, making that a `TryGet` would be more inline with how other cache APIs work.